src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -221,4 +222,16 @@ public static String parseDescription(String description) throws ParseException
         }
         return trimmedDescription;
     }
+
+    /**
+     * Parses a {@code String} into a {@code Goal.Frequency}
+     *
+     * @throws ParseException if the given {@code String} is an invalid frequency.
+     */
+    public static Goal.Frequency parseFrequency(String frequencyString) throws ParseException {
+        if (!Goal.ENUM_MAP.containsKey(frequencyString)) {
```

Comment:
Can we encapsulate the `ENUM_MAP` so we could change implementation later on? `Goal.isValidFrequencyString(frequencyString)` would be a better API